tests/zfs-tests/tests/functional/cli_root/zpool_add/zpool_add_010_pos.ksh

```diff
@@ -0,0 +1,166 @@
+#!/bin/ksh -p
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright 2009 Sun Microsystems, Inc.  All rights reserved.
+# Use is subject to license terms.
+#
+
+#
+# Copyright (c) 2012, 2016 by Delphix. All rights reserved.
+#
+
+. $STF_SUITE/include/libtest.shlib
+. $STF_SUITE/tests/functional/cli_root/zpool_create/zpool_create.shlib
+
+#
+# DESCRIPTION:
+#	Verify zpool add succeed when adding vdevs with matching redundancy.
+#
+# STRATEGY:
+#	1. Create base filesystem to hold virtual disk files.
+#	2. Create several files == $MINVDEVSIZE.
+#	3. Create pool with given redundancy.
+#	3. Verify 'zpool add' succeed with with matching redundancy.
+#
+
+verify_runnable "global"
+
+function cleanup
+{
+	datasetexists $TESTPOOL1 && destroy_pool $TESTPOOL1
+	datasetexists $TESTPOOL && destroy_pool $TESTPOOL
+}
+
+
+log_assert "Verify 'zpool add' succeed with keywords combination."
+log_onexit cleanup
+
+create_pool $TESTPOOL $DISKS
+mntpnt=$(get_prop mountpoint $TESTPOOL)
+
+typeset -i i=0
+while ((i < 10)); do
+	log_must mkfile $MINVDEVSIZE $mntpnt/vdev$i
```

Comment:
You can use `truncate -s` here instead of `mkfile` to more quickly create a sparse file.  It looks like that's shave 11-12 seconds of the test time.  We could make this same tweak to `zpool_create_006` too.